### PR TITLE
Auth/pm-47

### PR DESCRIPTION
**Pull request by dgoodman-bw**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
This PR contains the mapping of roles and their permissions, and the foundational logic of the RBAC implementation. 

## Code changes

* **AccessControlService.cs** A function `UserHasPermission(Permission)` has been added that serves as the core of the RBAC logic. This function will be called in the Views to determine whether to display certain UI elements, and will be used in the ActionFilterAttribute that will be implemented in the next step.
* **Permissions.cs** List of enums that represent all current permissions.
* **RolePermissionMapping.cs** Contains a dictionary of Roles:Permissions. This is used to determine if a user has a given permission (their role is used as the key, and if the permission is in the list of values for that key, they have the permission). In the next iteration of the RBAC implementation, this file will be replaced with a db call to retrieve the permissions for a given role.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Admin/Services/AccessControlService.cs [hunk 1]
```diff
@@ -17,6 +21,18 @@ public class AccessControlService : IAccessControlService
         _globalSettings = globalSettings;
     }
 
+    public bool UserHasPermission(Permission permission)
+    {
+        if (_globalSettings.SelfHosted)
+            return true;
+
+        var userRole = GetUserRoleFromClaim();
```
Comment (trmartin4) [inline src/Admin/Services/AccessControlService.cs:29]: Is the assumption here that a user can only be in one role at a time?

File: src/Admin/Utilities/RolePermissionMapping.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
+﻿using Bit.Admin.Enums;
+
+namespace Bit.Admin.Utilities;
+
+public static class RolePermissionMapping
+{
+    public static readonly Dictionary<string, List<Permission>> RolePermissions = new Dictionary<string, List<Permission>>()
```
Comment (trmartin4) [inline src/Admin/Utilities/RolePermissionMapping.cs:7]: Definitely looking forward to this being in a table!

**Review by trmartin4 — COMMENTED** [review]
Just a few questions, but it looks good!

File: src/Admin/Services/AccessControlService.cs (hunk 1 above)
Comment (dgoodman-bw) [reply src/Admin/Services/AccessControlService.cs:29]: Yes! The requirement is for a user to only ever have one role.

File: src/Admin/Services/AccessControlService.cs [hunk 3]
```diff
@@ -17,6 +21,18 @@ public class AccessControlService : IAccessControlService
         _globalSettings = globalSettings;
     }
 
+    public bool UserHasPermission(Permission permission)
+    {
+        if (_globalSettings.SelfHosted)
+            return true;
```
Comment (jlf0dev) [inline src/Admin/Services/AccessControlService.cs:27]: we try to always [use curly braces for control blocks](https://contributing.bitwarden.com/contributing/code-style/csharp#control-blocks)

File: src/Admin/Services/AccessControlService.cs [hunk 4]
```diff
@@ -17,6 +21,18 @@ public class AccessControlService : IAccessControlService
...
+
+        var userRole = GetUserRoleFromClaim();
+        if (string.IsNullOrEmpty(userRole) || !RolePermissionMapping.RolePermissions.ContainsKey(userRole))
+            return false;
```
Comment (jlf0dev) [inline src/Admin/Services/AccessControlService.cs:31]: curly braces 😁

File: src/Admin/Utilities/RolePermissionMapping.cs (hunk 2 above)
Comment (jlf0dev) [reply src/Admin/Utilities/RolePermissionMapping.cs:7]: suggestion: a TODO here might be nice to mark this as temporary.

**Review by jlf0dev — CHANGES_REQUESTED** [review]
Just a couple notes but it's looking good! As someone who's been dealing with permissions as well it's nice to see some work being done on them!

File: src/Admin/Services/AccessControlService.cs (hunk 4 above)
Comment (dgoodman-bw) [reply src/Admin/Services/AccessControlService.cs:34]: Thank you! I've fixed this and some other instances

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**